cloudwatch.tf

```diff
@@ -46,3 +46,47 @@ resource "aws_cloudwatch_metric_alarm" "cache_memory" {
 }
 */
 
+resource "aws_cloudwatch_metric_alarm" "redis-elasticache-high-db-memory-warning" {
+  count = "${var.redis_clusters}"
+
+  alarm_name          = "alarm-${var.name}-${local.vpc_name}-CacheCluster00${count.index + 1}-DatabaseMemoryUsagePercentage-90-percent"
+  alarm_description   = "Alert triggered when Redis usage is greater than or equal to 90%."
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = "DatabaseMemoryUsagePercentage"
+  namespace           = "AWS/ElastiCache"
+  period              = "300"
+  statistic           = "Average"
+
+  threshold           = "${var.p2_redis_threshold}"
+
+  alarm_actions       = ["${var.sns_topic_arn}"]
+  ok_actions          = ["${var.sns_topic_arn}"]
+
+  dimensions {
+    CacheClusterId = "${aws_elasticache_replication_group.redis.id}-00${count.index + 1}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "redis-elasticache-high-db-memory-critical" {
+  count = "${var.redis_clusters}"
+
+  alarm_name          = "alarm-${var.name}-${local.vpc_name}-CacheCluster00${count.index + 1}-DatabaseMemoryUsagePercentage-96-percent"
```

Comment:
This alarm_name could be more descriptive and follow a consistent naming convention.
Also consider making '96' here dynamic, as you have done with 'var.p1_redis_threshold'.
Perhaps a local could be used to ensure the alarm name and threshold are always in sync.